osv/impact.py

```diff
@@ -337,23 +338,54 @@ def _throttled_delete(to_delete):
       time.sleep(_DATASTORE_BATCH_SLEEP)
 
 
-def update_affected_commits(bug_id, commits, public):
+def update_affected_commits(bug_id, commits, public, write_legacy=True):
   """Update affected commits."""
   to_put = []
   to_delete = []
 
-  for commit in commits:
-    affected_commit = models.AffectedCommit(
-        id=bug_id + '-' + commit, bug_id=bug_id, commit=commit, public=public)
-
-    to_put.append(affected_commit)
-
-  # Delete any affected commits that no longer apply. This can happen in cases
-  # where a FixResult comes in later and we had previously marked a commit prior
-  # to the fix commit as being affected by a vulnerability.
-  for existing in models.AffectedCommit.query(
-      models.AffectedCommit.bug_id == bug_id):
-    if existing.commit not in commits:
+  # TODO(ochang): Remove this completedly once AffectedCommits migration is
+  # done.
+  if write_legacy:
+    for commit in commits:
+      affected_commit = models.AffectedCommit(
+          id=bug_id + '-' + commit, bug_id=bug_id, commit=commit, public=public)
+
+      to_put.append(affected_commit)
+
+    # Delete any affected commits that no longer apply. This can happen in
+    # cases where a FixResult comes in later and we had previously marked a
+    # commit prior to the fix commit as being affected by a vulnerability.
+    for existing in models.AffectedCommit.query(
+        models.AffectedCommit.bug_id == bug_id):
+      if existing.commit not in commits:
+        to_delete.append(existing.key)
+
+    _throttled_put(to_put)
+    _throttled_delete(to_delete)
+
+  to_put.clear()
```

Comment:
nit: these can be indented to be part of the if statement

Reply:
I'm trying to determine how nitty this actually is. Which if statement?

Reply:
Done!